java/ql/lib/ext/retrofit2.model.yml

```diff
@@ -4,3 +4,8 @@ extensions:
       extensible: sinkModel
     data:
       - ["retrofit2", "Retrofit$Builder", True, "baseUrl", "", "", "Argument[0]", "request-forgery", "manual"]
+  - addsTo:
+      pack: codeql/java-all
+      extensible: summaryModel
+    data:
+      - ["retrofit2", "Retrofit$Builder", False, "baseUrl", "(String)", "", "Argument[this]", "ReturnValue", "taint", "ai-manual"]
```

Comment:
We should try to avoid having summaries and sinks at the same node (see line 6).

Reply:
Wouldn't this mean that the example vulnerability below is no longer found?

```
db.query("SELECT * FROM users WHERE webpage = " + builder.toString());
```

Reply:
Yes, that's one of the risks of applying the "either sink or summary" rule. In the real world, though, how often one is going to use an HTTP request builder to do something else (e.g. build a SQL query)? Does your example come from a real world application, or is it a what if?

Reply:
(Not saying that we shouldn't change the rule, I always struggle myself with deciding what to do in these cases)

Reply:
It's a 'what if', but it doesn't seem that contrived to me. storing urls in a database isn't a crazy idea..

> "either sink or summary" rule

I know nothing of such a rule in the first place.. where does this come from? how strict is this rule?

Reply:
> I know nothing of such a rule in the first place..
Oh, I thought we mentioned it during some triage session.

> where does this come from? how strict is this rule?
Originally, it was to prevent having duplicated/overlapping paths in the form of:

1. `src` -> `sink/summary`
2. `src` -> `sink/summary` -> `sink`

About how strict it is... We get to decide 😄. This isn't written anywhere nor I think anyone else is applying it. If experience and real-world results tell us that it doesn't make sense, we should get rid of it. In this case I see the argument for keeping both models.

I'm going to review the alert diff in DCA to make sure but, if nothing looks wrong, we can go ahead as it is.

Reply:
> Oh, I thought we mentioned it during some triage session.

The name was dropped and I briefly reacted but we continued the triage w/o me really grokking where the rule is coming from. For the record, I'm happy with or without the rule — I have no horse in this race, I only want to understand the current way to model.

> I'm going to review the alert diff in DCA to make sure but, if nothing looks wrong, we can go ahead as it is.

Thank you!